rasa/utils/common.py

```diff
@@ -504,3 +504,14 @@ def find_unavailable_packages(package_names: List[Text]) -> Set[Text]:
 def module_path_from_class(clazz: Type) -> Text:
     """Return the module path of an instance's class."""
     return clazz.__module__ + "." + clazz.__name__
+
+
+def get_bool_env_variable(variable_name: str, default_variable_value: bool) -> bool:
+    true_values = ("true", "1")
+    false_values = ("false", "0")
+    value = os.getenv(variable_name, None)
+    if value is None:
+        value = str(default_variable_value)
+    if value.lower() not in true_values + false_values:
+        raise ValueError(f"Invalid value `{value}` for variable `{variable_name}`")
```

Comment:
I would refactor this to:
```suggestion
    if not isinstance(value, bool) or (isinstance(value, int) and value not in [0,1]):
        raise ValueError(f"Invalid value `{value}` for variable `{variable_name}`")
```
I would also make the error message more explainable (define expected data type/values that the env variable can take). Also since this would be a user error, this should be a `RasaException` not a ValueError.

Reply:
It is not possible for value to be of type _bool_ or _int_ because **os.getenv** returns _str | T (bool)_ if default value is boolean.
This is why I have limited the code to just use strings, because it makes the implementation smaller and it still produces same results.